src/oncall/user_sync/ldap_sync.py

```diff
@@ -242,7 +244,9 @@ def sync(config, engine):
                 engine.execute(name_update_sql, (full_name, username))
                 stats['user_names_updated'] += 1
             if not db_contacts.get('photo_url'):
-                engine.execute(photo_update_sql, (LDAP_SETTINGS['image_url'] % username, username))
+                photo_url_tpl = LDAP_SETTINGS.get('image_url')
+                photo_url = photo_url_tpl % username if photo_url_tpl else None
```

Comment:
If photo_url is NULL in the database, this will execute the UPDATE query on every row. Instead, use `if 'image_url' in LDAP_SETTINGS and not db_contacts.get('photo_url'):` on line 246.